src/traces/ohlc/helpers.js

```diff
@@ -97,13 +97,46 @@ exports.makeTransform = function(traceIn, state, direction) {
 exports.getFilterFn = function(direction) {
     switch(direction) {
         case 'increasing':
-            return function(o, c) { return o <= c; };
+            return function(o, c, isPrevThisDirection, oprev, cprev) {
```

Comment:
The algo looks good :+1: 

It might worth the time to make it a little DRYer by wrapping `isPrevThisDirestion`, `oprev` and `cprev` in a closure. That is, something like:

```js
exports.getFilterFn = function(direction) {
  return new _getFilterFn(direction);
};

function _getFilterFn(directon) {
  var isPrevThisDirection = null;
  var oprev = null;
  var cprev = null;
  
  var fn;
  switch(direction) {
    case 'increasing':
      fn = function(o, c) { /* */ };
      break;
    case 'decreasing':
      fn = function(o, c) { /* */ };
      break;
  }

  return function(o, c) {
    var out = fn(o, c);
    isPrevThisDirection = !!out;
    oprev = o;
    cprev = c;
  };
};
```

that way we wouldn't have to patch `candlestick/transform.js` and `ohlc/transform.js`.